driver/annotations.md

```diff
@@ -0,0 +1,171 @@
+  the predicate(s) used as argument.
+
+- `HasInternalType(type string)`: checks if the InternalType of the node matches
+  the string provided. What key is the InternalType of a native AST node is 
+  defined in the `parser.ObjectToNoder.InternalTypeKey` field.
+
+- `HasToken(name string)`: checks if the node has a token matching the given
+  string. What key of the node is a Token is defined in the 
+  `parser.ObjectToNoder.TokenKeys` map.
+
+- `HasChild(predicate)`: will check if any of the first level children of
+  the current node matches the given predicate.
+
+- `HasProperty(key, value string)`: checks if the AST node has some specific key
+  and the key as the given value.
+
+- `HasInternalRole(key string)`: this is actually a very handy shortcut to 
+  `HasProperty(uast.InternalRoleKey, value)` that will check if the node has
+  an InternalRole matching the specified key. InternalRoles are
+  added to children of any object keys holding them.
+
+#### Actions
+
+- `Roles(roles... uast.Role)`: Will add the given UAST role to the currently
+  selected node(s).
+
+- `Error(err Error)`: Will produce an error and stop processing the AST.
+
+#### Selectors
+
+- `Self(rules ...*Rule)`: Selects the current node.
+
+- `Children(rules ...*Rule)`: Select the children of the current node. Please note
+   that only the first level children are selected, second level ("grandchildren")
+   and others below are not selected (use Descendants for those).
+
+- `Descendants(rules ...*Rule)`: Selects all (recursive) descendants of the
+  selected node.
+
+- `DescendantsOrSelf(rules ...*Rule)`: Selects the current node and all
+  (recursive) descendants.
+
+### Examples:
+
+Check the already implemented [Java
+annotator](https://github.com/bblfsh/java-driver/blob/master/driver/normalizer/annotation.go)
+or the [Python
+annotator](https://github.com/bblfsh/python-driver/blob/master/driver/normalizer/annotation.go)
+for real world examples of this API in action.
+
+### When in doubt...
+
+[//]: # FIXME: (uncomment this paragraph when the table is available)
+[//]: # If you find a node in the native AST that you're unsure how to map into the UAST
+[//]: # the first step is to check how other languages map the same or similar nodes.  For
+[//]: # this you can check this [Annotation table](FIXME link when available) that show how
+[//]: # different languages have mapped their internal roles to the UAST ones.
+
+You can also ask any doubt on the project's public [Babelfish Slack
+channel](https://sourced-community.slack.com/join/shared_invite/MTkwNTM0ODEyODIzLTE0OTYxMzc5NTMtODRhMDYyNzAyYQ)
+which is very friendly to newcomers to the project.
+
+Finally, if you really think that there isn't a correspondence in the UAST roles
+for the native role that you want to map, you can fork the [Babelfish SDK project
+on Github](https://github.com/bblfsh/sdk), add the new role to the file
+`uast/uast.go` and make a PR. Don't expect the role to be added inmediatedly;
```

Comment:
This is after the text about asking in the Slack channel and will be also after the doc about the table @alcortesm is doing so it's really the last option, but I'll add a sentence about opening an issue as alternative to the PR. Once the BIPs are merged I would link those too.